onelogin-aws-assume-role-cli/src/main/java/com/onelogin/aws/assume/role/cli/OneloginAWSCLI.java

```diff
@@ -24,149 +42,307 @@
+			commandLine = cmd.parse(options, commandLineArguments);
+			String value;
+			if (commandLine.hasOption("time")) {
+				value = commandLine.getOptionValue("time");
+				if (value != null && !value.isEmpty()) {
+					time = Integer.parseInt(value);
+				}
+				if (time < 15 ) {
+					time = 15;
+				}
+				if (time > 60 ) {
+					time = 60;
+				}
+			}
+			if (commandLine.hasOption("loop")) {
+				value = commandLine.getOptionValue("loop");
+				if (value != null && !value.isEmpty()) {
+					loop = Integer.parseInt(value);
+				}
+			}
+			if (commandLine.hasOption("profile")) {
+				value = commandLine.getOptionValue("profile");
+				if (value != null && !value.isEmpty()) {
+					profileName = value;
+				} else {
+					profileName = "default";
+				}
+			}
+			if (commandLine.hasOption("file")) {
+				value = commandLine.getOptionValue("file");
+				if (value != null && !value.isEmpty()) {
+					file = new File(value); 
+				}
+			}
+		} catch (ParseException parseException) {
+			System.err.println("Encountered exception while parsing" + parseException.getMessage());
+		}
+	}
+
+	public static Options buildOptions() {
+		final Options options = new Options();
+		options.addOption("t", "time", true, "Sleep time between iterations, in minutes  [15-60 min]");
+		options.addOption("l", "loop", true, "Number of iterations");
+		options.addOption("p", "profile", true, "Save Temporal AWS credentials using that profile name");
+		options.addOption("f", "file", true, "Set a custom path to save the AWS credentials. (if not used, default path is used)");
+		return options;
+	}
+
+	public static void main(String[] commandLineArguments) throws Exception {
+		commandParser(commandLineArguments);
+
+		// OneLogin Java SDK Client
+		Client olClient = new Client();
+		olClient.getAccessToken();
+		Scanner scanner = new Scanner(System.in);
+		try {
+			String oneloginUsernameOrEmail = null;
+			String oneloginPassword = null;
+			String appId = null;
+			String oneloginDomain = null;
+			String samlResponse;
+			String awsRegion = null;
+
+			Map<String, String> mfaVerifyInfo = null;
+			Map<String, Object> result;
+
+			String roleArn = null;
+			String principalArn = null;
+			String defaultAWSRegion = Regions.DEFAULT_REGION.getName();
+
+			for (int i = 0; i < loop; i++) {
+				if (i == 0) {
+					// Capture OneLogin Account Details
+					System.out.print("OneLogin Username: ");
+					oneloginUsernameOrEmail = scanner.next();
+					System.out.print("OneLogin Password: ");
+					try {
+						oneloginPassword = String.valueOf(System.console().readPassword());
+					} catch (Exception e) {
+						oneloginPassword = scanner.next();
+					}
+					System.out.print("AWS App ID: ");
+					appId = scanner.next();
+					System.out.print("Onelogin Instance Sub Domain: ");
+					oneloginDomain = scanner.next();
+				} else {
+					TimeUnit.MINUTES.sleep(time);
+				}
+
+				result = getSamlResponse(olClient, scanner, oneloginUsernameOrEmail, oneloginPassword, appId,
```

Comment:
I havent tested this but I assume the user will be prompted for MFA on each loop?

Reply:
Yes, since MFA token has a very short expiration...

Reply:
👍